src/SFML/Graphics/Image.cpp

```diff
@@ -208,6 +208,10 @@ void Image::copy(const Image& source, const Vector2u& dest, const IntRect& sourc
     if ((source.m_size.x == 0) || (source.m_size.y == 0) || (m_size.x == 0) || (m_size.y == 0))
         return;
 
+    // Make sure the sourceRect is valid
+    if (sourceRect.left < 0 || static_cast<unsigned>(sourceRect.left) >= source.m_size.x || sourceRect.top < 0 || static_cast<unsigned>(sourceRect.top) >= source.m_size.y)
+        return;
+
```

Comment:
Maybe it's worthwhile splitting this in a separate PR, and having `Image::copy` return a `[[nodiscard]] bool` so that users can actually check the copy happened. Thoughts? @eXpl0it3r

Reply:
Maybe worth just having this PR with a commit of tests, and a commit of the fix go in as one. Then a future PR to revisit the `sf::Image::copy()` to mark it as `[[nodiscard]] bool`. At least that way we get the fix in sooner and the tests are also in to verify future changes to `sf::Image`

Reply:
Let's extend the comment by briefly mentioning why we only have to check the top-left corner